### PR TITLE
Update to build with gh-8.2.1 and Cabal-2.0.0.2

### PR DESCRIPTION
**Pull request by judah**
Also add `stack-8.2.1.yaml` and track this in Travis CI.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -43,7 +47,8 @@ install:
   - case "$BUILD" in
       stack)
         $STACK setup --no-terminal;
-        $STACK build --only-snapshot --no-terminal;;
+        $STACK build --only-snapshot --no-terminal;
+        $STACK build --no-terminal happy;;
```
Comment (blackgnezdo) [inline .travis.yml]: Does ;; have to be at the end of the last line? I'm looking for a way to format this such that I'd only see a single line diff, not 2 lines.

File: lens-labels/src/Lens/Labels.hs [hunk 2]
```diff
@@ -81,8 +84,12 @@ class HasLens (x :: Symbol) f s t a b
 instance
     (p ~ (a -> f b), q ~ (s -> f t), HasLens x f s t a b)
     => IsLabel x (LensFn p q) where
+#if __GLASGOW_HASKELL__ >= 802
+    fromLabel = LensFn $ lensOf (proxy# :: Proxy# x)
```
Comment (blackgnezdo) [inline lens-labels/src/Lens/Labels.hs]: What are the trailing # hash thingies for?

File: stack-8.2.1.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Build setup for ghc-8.2.1.
```
Comment (blackgnezdo) [inline stack-8.2.1.yaml]: I take it git doesn't recognize this as derived from stack.yaml?

File: stack-8.2.1.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# TODO: remove this file after Stackage releases an LTS for this GHC version.
+# of GHC.
+setup-info:
+  ghc:
+    windows64:
```
Comment (blackgnezdo) [inline stack-8.2.1.yaml]: Are you using this?!

File: stack-8.2.1.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+- aeson-1.2.1.0
+- array-0.5.2.0
+- base-orphans-0.6
+- criterion-1.2.1.0
+- deepseq-1.4.3.0
+- filepath-1.4.1.2
+- microstache-1.0.1.1
+- primitive-0.6.2.0
+- statistics-0.14.0.2
+- unix-2.7.2.2
```
Comment (blackgnezdo) [inline stack-8.2.1.yaml]: Doesn't seem to jive with windows64 above?

File: stack-8.2.1.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+# Build setup for ghc-8.2.1.
+# TODO: remove this file after Stackage releases an LTS for this GHC version.
+# of GHC.
+setup-info:
+  ghc:
+    windows64:
+      8.2.1:
+        url: https://downloads.haskell.org/~ghc/8.2.1/ghc-8.2.1-x86_64-unknown-mingw32.tar.xz
+        sha1: bdede26c1a2cfcf4ebb08329853d285e32213b3d
```
Comment (blackgnezdo) [inline stack-8.2.1.yaml]: sha256 not an option yet?

File: .travis.yml (hunk 1 above)
Comment (judah) [inline .travis.yml]: Yeah, I think this is a bash-ism for case statements (or at least how Travis uses it).

File: lens-labels/src/Lens/Labels.hs (hunk 2 above)
Comment (judah) [inline lens-labels/src/Lens/Labels.hs]: It's `-XMagicHash`:
https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#the-magic-hash

File: stack-8.2.1.yaml (hunk 3 above)
Comment (judah) [inline stack-8.2.1.yaml]: Yeah, git doesn't have a way to track "copy" actions.

File: stack-8.2.1.yaml (hunk 4 above)
Comment (judah) [inline stack-8.2.1.yaml]: Removed; I copied it from other instructions, but as you point out it's not tested (and easy enough for other devs to add).

File: stack-8.2.1.yaml (hunk 6 above)
Comment (judah) [inline stack-8.2.1.yaml]: Stack didn't seem to recognize that as field.

File: stack-8.2.1.yaml (hunk 5 above)
Comment (judah) [inline stack-8.2.1.yaml]: Removed windows64.

**Pull request merged**